### PR TITLE
[move] sui framework cleanup

### PR DESCRIPTION
**Pull request by sblackshear**
- Moving `collection`, `bag`, `object_basics` to examples
- Deleting redundant implementations of `transfer` on types with `store`
- Fixing the numerous tests that expect `object_basics` to be part of the framework :/

### CODE REVIEW

File: crates/sui-adapter-transactional-tests/tests/sui/freeze.move [hunk 1]
```diff
@@ -3,12 +3,76 @@
 
 // test that freezing prevents transfers/mutations
 
-//# init --accounts A
+//# init --addresses test=0x0 --accounts A
```
Comment (sblackshear) [inline crates/sui-adapter-transactional-tests/tests/sui/freeze.move:6]: @tnowacki, could you help me figure out how to fix this test? I am copying what is done in the other publishing tests and think I did it right, but am getting

```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: DuplicateObjectRefInput', crates/sui-transactional-test-runner/src/test_adapter.rs:472:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Review by damirka — APPROVED** [review]
Thanks!

**Comment by damirka** [comment]
Please check documentation as well. There might be links to any of the modules (I assume object_basics is the one that is referenced there)

Cc @Clay-Mysten

File: crates/sui-adapter-transactional-tests/tests/sui/freeze.move (hunk 1 above)
Comment (sblackshear) [reply crates/sui-adapter-transactional-tests/tests/sui/freeze.move:6]: NVM, I figured it out--just needed to update those object #'s.

**Comment by Clay-Mysten** [comment]
> Please check documentation as well. There might be links to any of the modules (I assume object_basics is the one that is referenced there)
> 
> Cc @Clay-Mysten

Yes, thank you @damirka! @sblackshear can we go over the effects of this together? I am wondering if there are changes needed beyond links, as well, as as the switch from `transfer` to `split_and_transfer` here. I assume that is not in all cases but want to confirm. We don't seem to mention it now:
```
/Users/claymurphy/vscode/sui
claymurphy@Clays-MacBook-Pro sui % grep -r -i split_and_transfer doc
claymurphy@Clays-MacBook-Pro sui % 
```
Thanks!

**Comment by sblackshear** [comment]
@Clay-Mysten on docs, I would not expect this to have any effects. I took a careful look at this and:

- There are no links to fix
- There is no switch from `coin::transfer` to `coin::split_and_transfer`, we are simply removing `coin::transfer`. We have already been asking folks to use `transfer::transfer` for some time (which most code examples/tests already do), this is just removing one "wrong" way to do things.
- `object_basics` is not mentioned in any docs except in a (harmless) example of a log message where the name is not meaningful
- `bag` and `collection` are not used in docs
- `sui::transfer` is used in a tutorial, so I left it in for now. Later, we should remove this and update the tutorial, but I don't want to add that to this already-large PR
- There are no other uses of `::transfer` (e.g., `coin::transfer` or `sui::transfer`) except for the expected `transfer::transfer` and `transfer::transfer_to_object`

I am planning to land this as soon as I can because there are many potential conflicts with tests and genesis. But please let me know if you have concerns beyond the above and we can discuss!

**Review by awelc — COMMENTED** [review]
LGTM

**Comment by Clay-Mysten** [comment]
@sblackshear I should replace here though, correct?:
---
coin::transfer
---

**Comment by sblackshear** [comment]
> @sblackshear I should replace here though, correct?:
> 
> ```
> claymurphy@Clays-MacBook-Pro sui % grep -r -i "coin::transfer" doc  
> doc/src/contribute/cli-client.md:    coin::transfer(c, Address::new(recipient))
> claymurphy@Clays-MacBook-Pro sui % 
> ```

We could replace this, but the snippet is already out-of-date on other ways (e.g., `Address::new(recipient)` is no longer needed/will not work). I think it would be best for us to take on the bigger task of refactoring `cli-client.md` to call a function other than `sui::transfer`--perhaps `coin::split_and_transfer` would be a good candidate.

**Pull request merged**